onebusaway-android/src/main/java/com/joulespersecond/seattlebusbot/GlassArrivalsListActivity.java

```diff
@@ -0,0 +1,719 @@
+                Log.d(TAG,
+                        "Region info has expired (or does not exist), forcing a reload from the server...");
+            }
+        }
+
+        if (Application.get().getCurrentRegion() != null) {
+            Log.d(TAG, "Using previous region: " + Application.get().getCurrentRegion().getName());
+
+            if (mLastKnownLocation != null) {
+                getClosestStops(mLastKnownLocation);
+            } else {
+                Log.d(TAG,
+                        "Couldn't get user's location to find nearby stops - will wait for location");
+                return;
+            }
+        }
+
+        if (mLastKnownLocation != null) {
+            //Check region status, possibly forcing a reload from server and checking proximity to current region
+            //Normal progress dialog doesn't work, so hard-code false as last argument
+            mObaRegionsTask = new ObaRegionsTask(this, this, forceReload, showProgressDialog);
+            mObaRegionsTask.execute();
+        } else {
+            Log.d(TAG,
+                    "Couldn't get user's location to find closest region - will wait for location");
+            return;
+        }
+    }
+
+    private void getClosestStops(Location location) {
+        // Update progress message
+        mProgressMessage.setText(getString(R.string.finding_closest_stop));
+
+        // Find the closest stops
+        mObaStopsForLocationTask = new ObaStopsForLocationTask(this, this, location);
+        mObaStopsForLocationTask.execute();
+    }
+
+    private void showErrorMessage(String errorMessage) {
+        mDistanceToStopView.setVisibility(View.GONE);
+        mArrowView.setVisibility(View.GONE);
+
+        mProgressMessage.setText(errorMessage);
+        mIndeterm.stopIndeterminate();
+        mIndeterm.stopProgress();
+    }
+
+    /*
+     * Callbacks from tasks
+     */
+
+    // For Oba Regions Task
+    @Override
+    public void onTaskFinished(boolean currentRegionChanged) {
+        if (currentRegionChanged) {
+            Log.d(TAG, "New region selected - now finding stops...");
+            // The current region has changed since last startup, so abort any existing stop
+            // request in progress (which would be using the previous region API) and start a new one
+            if (mObaStopsForLocationTask != null) {
+                mObaStopsForLocationTask.cancel(true);
+            }
+
+            getClosestStops(mLastKnownLocation);
+        }
+    }
+
+    // For StopsForLocation Request
+    @Override
+    public void onTaskFinished(ObaStopsForLocationResponse response) {
+        if (response.getCode() != ObaApi.OBA_OK) {
+            showErrorMessage(getString(UIHelp.getStopErrorString(this, response.getCode())));
+            return;
+        }
+
+        Log.d(TAG, "Found stops.");
+        // Find closest stop
+        ObaStop closestStop;
+        closestStop = UIHelp.getClosestStop(this, response.getStops(), mLastKnownLocation);
```

Comment:
Agreed!  Code in this PR isn't my best work :).  It's more of a shortest path to get something working on Glass.